File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        Deque<Integer> queue = new ArrayDeque<>();

        StringTokenizer commamd;

        int n = Integer.parseInt(br.readLine());
        for (int i=0;i<n;i++) {
            commamd = new StringTokenizer(br.readLine());

            switch (commamd.nextToken()) {
                case "push":  // 뒤쪽에 데이터 삽입
                    queue.add(Integer.parseInt(commamd.nextToken()));
                    break;
                case "pop" :  // 앞쪽의 데이터 삭제
                    if (queue.size()==0) {
                        sb.append(-1).append('\n');
                    } else {
                        sb.append(queue.getFirst()).append('\n');
                        queue.remove();
                    }
                    break;
                case "size":
                    sb.append(queue.size()).append('\n');
                    break;
                case "empty":
                    if(queue.size()==0) {
                        sb.append(1).append('\n');
                    } else
                        sb.append(0).append('\n');
                    break;
                case "front":
                    if (queue.size()==0) {
                        sb.append(-1).append('\n');
                    } else
                        sb.append(queue.getFirst()).append('\n');
                    break;
                case "back":
                    if (queue.size()==0) {
                        sb.append(-1).append('\n');
                    } else
                        sb.append(queue.getLast()).append('\n');
                    break;
            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque
input = sys.stdin.readline

n = int(input())
q = deque()

for i in range(n):
    command = input().strip()
    if command.startswith("push"):
        cmd, val = command.split()
        q.append(val)
    elif command == "pop":
        print(q.popleft() if q else -1)
    elif command == "size":
        print(len(q))
    elif command == "empty":
        print(1 if not q else 0)
    elif command == "front":
        print(q[0] if q else -1)
    elif command == "back":
        print(q[-1] if q else -1)